README.md

```diff
@@ -1,54 +1,190 @@
+[`ExchangeGrantTokenJob`](app/jobs/exchange_grant_token_job.rb) with the
+`sandwich_id`. That job calls the
+[`GrantCodeExchanger`](app/services/grant_code_exchanger.rb) service class,
+which sends a POST request to `https://id.heroku.com/oauth/token` with the grant
+code. The Platform API for Partners responds with the `refresh_token` and
+`access_token`. An example of this response is included in [the
+fixtures](spec/support/fixtures/grant_code_exchange_response.json).
 
-Install and start PostgreSQL.
+`GrantCodeExchanger` saves the `access_token`, `refresh_token`, and an
+expiration timestamp for the `access_token` on the Sandwich record. The
+`access_token` and `refresh_token` are encrypted at rest using the
+[`attr_encrypted`](https://github.com/attr-encrypted/attr_encrypted) gem. Once the
+`access_token` is expired, a new one can be obtained using the `refresh_token`.
 
-```
-brew install postgresql
-brew services start postgresql
-```
+In the Sudo Sandwich app, we use the `access_token` for async provisioning. The
+`access_token` is also required for accessing other endpoints in the Platform
+API for Partners. Those endpoints are not accessed in Sudo Sandwich but you can
+learn more about them by reading [the Platform API for Partners
+documentation](https://devcenter.heroku.com/articles/platform-api-for-partners#list-of-standard-platform-api-endpoints).
 
-[Homebrew]: https://brew.sh/
-[Heroku CLI]: https://devcenter.heroku.com/articles/heroku-cli
+## The deprovisioning request
+
+### Heroku docs
 
-### Ruby on Rails
+* https://devcenter.heroku.com/articles/building-an-add-on#the-deprovisioning-request
 
-This application is built using [Ruby on Rails].
+### Related code
 
-Your system will require [Ruby] to develop on the application.
+Deprovisioning happens via the
+[`Heroku::ResourcesController`](app/controllers/heroku/resources_controller.rb)
+`#destroy` method. The controller deletes the Sandwich record that corresponds with the
+`heroku_uuid` sent with the request and returns a status of `204` to indicate
+that the request was successfully received and processed.
 
-The required Ruby version is listed in the [.ruby-version](.ruby-version) file.
+## The plan change request
 
-If you do not have this binary, [use this guide to get set up on MacOS].
+### Heroku docs
+
+* https://devcenter.heroku.com/articles/building-an-add-on#the-plan-change-request-upgrade-downgrade
 
-[Ruby on Rails]: http://rubyonrails.org
-[Ruby]: https://www.ruby-lang.org/en/
-[use this guide to get set up on MacOS]: http://installfest.railsbridge.org/installfest/macintosh
+### Related code
+
+Plan changes happen via the
+[`Heroku::ResourcesController`](app/controllers/heroku/resources_controller.rb)
+`#update` method. The controller updates the plan of the Sandwich record that corresponds with the
+`heroku_uuid` sent with the request so that the plan matches the `plan` param
+sent with the request.
 
-### Configuring the Application
+## Single sign on
```

Comment:
maybe adding the term "Single Sign On Dashboard" would make this clearer?
(if that's true here, it's probably also true for the DevCenter article we link to, too)